### PR TITLE
feat(logging)_: enable log namespaces configuration

### LINKED ISSUE #6128: Enable logs filtering by namespace and level
Named logger instances currently lack individual configurability, limiting the ability to control logs based on specific needs. For example, when debugging missing messages, logs from unrelated components like the wallet are unnecessary and add noise.

#### Proposed Solution
Introduce a custom filtering mechanism for zap.Core to enable message filtering based on a specified format.

The proposed format for defining log levels (inspired by [go-log](https://github.com/ipfs/go-log/blob/d4825eba3c99b956004c578657a2df9ff08d55f8/README.md?plain=1#L68) from ipfs):
`<logger_name>:<level>[,<logger_name>:<level>...]`

Example:
`wallet:error,messenger:debug,waku:debug,messenger.communities:info`

### PR DESCRIPTION
**Pull request by osmaczko**
closes: #6128

### CODE REVIEW

File: tests-functional/tests/test_logging.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import re
+import time
+from test_cases import StatusBackend
+import pytest
+import os
+
+@pytest.mark.rpc
+@pytest.mark.skip("waiting for status-backend to be executed on the same host/container")
```
Comment (osmaczko) [inline tests-functional/tests/test_logging.py:8]: Skipped due to the issue described in comment, passing locally:

```cmd
Running pytest with args: ['-p', 'vscode_pytest', '--status_backend_urls=http://localhost:42995', '--rootdir=/path/to/status-go/tests-functional', '/path/to/status-go/tests-functional/tests/test_logging.py::TestLogging::test_logging']
============================= test session starts ==============================
platform linux -- Python 3.11.9, pytest-6.2.4, py-1.11.0, pluggy-0.13.1 -- /bin/python
cachedir: .pytest_cache
rootdir: /path/to/status-go/tests-functional, configfile: pytest.ini
plugins: dependency-0.6.0
collecting ... collected 1 item

tests/test_logging.py::TestLogging::test_logging 
-------------------------------- live log call ---------------------------------
INFO     root:status_backend.py:41 Sending POST request to url http://localhost:42995/statusgo/InitializeApplication with data: {
    "dataDir": "/tmp/pytest-of-mine/pytest-56/test_logging0"
}
INFO     root:signals.py:115 Connection opened
WARNING  websocket:_logging.py:66 websocket connected
INFO     root:status_backend.py:43 Got response: b'{"accounts":null,"centralizedMetricsInfo":{"enabled":false,"userConfirmed":false}}'
INFO     root:status_backend.py:49 Got response: b'{"accounts":null,"centralizedMetricsInfo":{"enabled":false,"userConfirmed":false}}'
INFO     root:status_backend.py:41 Sending POST request to url http://localhost:42995/statusgo/CreateAccountAndLogin with data: {
    "customizationColor": "primary",
    "displayName": "Mr_Meeseeks",
    "kdfIterations": 256000,
    "logEnabled": true,
    "logLevel": "INFO",
    "password": "Strong12345",
    "rootDataDir": "/tmp/pytest-of-mine/pytest-56/test_logging0"
}
INFO     root:status_backend.py:43 Got response: b'{"error":""}'
INFO     root:status_backend.py:49 Got response: b'{"error":""}'
INFO     root:rpc.py:51 Sending POST request to url http://localhost:42995/statusgo/CallRPC with data: {
    "id": null,
    "jsonrpc": "2.0",
    "method": "wakuext_setLogLevel",
    "params": [
        {
            "logLevel": "ERROR"
        }
    ]
}
INFO     root:rpc.py:55 Got response: {
    "id": null,
    "jsonrpc": "2.0",
    "result": null
}
INFO     root:rpc.py:51 Sending POST request to url http://localhost:42995/statusgo/CallRPC with data: {
    "id": null,
    "jsonrpc": "2.0",
    "method": "wakuext_setLogNamespaces",
    "params": [
        {
            "logNamespaces": "test1.test2:debug,test1.test2.test3:info"
        }
    ]
}
INFO     root:rpc.py:55 Got response: {
    "id": null,
    "jsonrpc": "2.0",
    "result": null
}
INFO     root:status_backend.py:41 Sending POST request to url http://localhost:42995/statusgo/Logout with data: {}
INFO     root:status_backend.py:43 Got response: b'{"error":""}'
INFO     root:status_backend.py:49 Got response: b'{"error":""}'
INFO     root:status_backend.py:41 Sending POST request to url http://localhost:42995/statusgo/LoginAccount with data: {
    "kdfIterations": 256000,
    "keyUid": "0x16a5754d8818c1bac5254d126d427248bd737109561b20a6162081721d358bb1",
    "password": "Strong12345"
}
INFO     root:status_backend.py:43 Got response: b'{"error":""}'
INFO     root:status_backend.py:49 Got response: b'{"error":""}'
INFO     root:rpc.py:51 Sending POST request to url http://localhost:42995/statusgo/CallRPC with data: {
    "id": null,
    "jsonrpc": "2.0",
    "method": "wakuext_logTest"
}
INFO     root:rpc.py:55 Got response: {
    "id": null,
    "jsonrpc": "2.0",
    "result": null
}
PASSED

============================== 1 passed in 12.21s ==============================
Finished running tests!
```

File: services/ext/api.go [hunk 2]
```diff
@@ -1808,10 +1808,29 @@ func (api *PublicAPI) SetLogLevel(request *requests.SetLogLevel) error {
...
 
+func (api *PublicAPI) SetLogNamespaces(request *requests.SetLogNamespaces) error {
+	return api.service.messenger.SetLogNamespaces(request)
+}
+
 func (api *PublicAPI) SetMaxLogBackups(request *requests.SetMaxLogBackups) error {
 	return api.service.messenger.SetMaxLogBackups(request)
 }
 
+func (api *PublicAPI) LogTest() error {
```
Comment (osmaczko) [inline services/ext/api.go:1819]: I couldn't find a more elegant solution. Initially, I reused logs that were already being printed during the flow, but this approach didn't cover all cases and made potential changes to the logs somewhat inflexible.

File: nodecfg/node_config.go [hunk 3]
```diff
@@ -89,10 +89,10 @@ func insertHTTPConfig(tx *sql.Tx, c *params.NodeConfig) error {
 func insertLogConfig(tx *sql.Tx, c *params.NodeConfig) error {
 	_, err := tx.Exec(`
 	INSERT OR REPLACE INTO log_config (
-		enabled, log_dir, log_level, max_backups, max_size,
+		enabled, log_dir, log_level, log_namespaces, max_backups, max_size,
 		file, compress_rotated, log_to_stderr, synthetic_id
-	) VALUES (?, ?, ?, ?, ?, ?, ?, ?, 'id'	)`,
-		c.LogEnabled, c.LogDir, c.LogLevel, c.LogMaxBackups, c.LogMaxSize,
+	) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, 'id')`,
+		c.LogEnabled, c.LogDir, c.LogLevel, c.LogNamespaces, c.LogMaxBackups, c.LogMaxSize,
```
Comment (igor-sirotin) [inline nodecfg/node_config.go:95]: Wanna rewrite it with Squirrel? 😏

File: tests-functional/tests/test_logging.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        # Init and login
+        backend_client.init_status_backend(data_dir=str(tmp_path))
+        backend_client.create_account_and_login(data_dir=str(tmp_path))
+        key_uid = self.ensure_logged_in(backend_client)
+
+        # Configure logging
+        backend_client.rpc_valid_request("wakuext_setLogLevel", [{"logLevel": "ERROR"}])
+        backend_client.rpc_valid_request("wakuext_setLogNamespaces", [{"logNamespaces": "test1.test2:debug,test1.test2.test3:info"}])
+
+        # Re-login (logging settings take effect after re-login)
```
Comment (igor-sirotin) [inline tests-functional/tests/test_logging.py:32]: hmm, why? I though there's an atomic setting now which applies right away 🤔

**Review by igor-sirotin — APPROVED** [review]
Amazing thing with the functional test 👍

File: tests-functional/tests/test_logging.py (hunk 4 above)
Comment (osmaczko) [reply tests-functional/tests/test_logging.py:32]: Yes, it is technically possible to set up logging at runtime. I followed the flow of `setLogLevel` for now, i.e., setLogLevel -> save settings in the database -> relogin -> load settings from database -> set up logging.

I'll address it in a follow-up to make it work at runtime and remove the popup about app restart requirement on desktop.

File: services/ext/api.go [hunk 5]
```diff
@@ -1808,10 +1808,28 @@ func (api *PublicAPI) SetLogLevel(request *requests.SetLogLevel) error {
 	return api.service.messenger.SetLogLevel(request)
 }
 
+func (api *PublicAPI) SetLogNamespaces(request *requests.SetLogNamespaces) error {
+	return api.service.messenger.SetLogNamespaces(request)
```
Comment (qfrank) [inline services/ext/api.go:1812]: it seems this API only update db, can we update corresponding namespaceFilteringCore instance as well?

Comment (osmaczko) [reply services/ext/api.go:1812]: Yes, please see: https://github.com/status-im/status-go/pull/6161#discussion_r1870082865

**Review by qfrank — APPROVED** [review]

**Pull request merged**